offset_index.go

```diff
@@ -23,21 +23,21 @@ type OffsetIndex interface {
 	FirstRowIndex(int) int64
 }
 
-type fileOffsetIndex format.OffsetIndex
+type FileOffsetIndex format.OffsetIndex
```

Comment:
Can we document this type? Especially mentioning how it's expected to be used by type checking the value returned by `OffsetIndex()`